imap/xapian_wrap.cpp

```diff
@@ -2451,17 +2629,34 @@ static Xapian::Query xapian_snipgen_build_query(xapian_snipgen_t *snipgen, Xapia
         }
     }
 
+    if (snipgen->numbers) {
+        for(size_t i = 0; i < snipgen->numbers->size(); ++i) {
```

Comment:
Here C++ syntax can be used, something like:
```c++
for (const auto &it : *(snipgen->numbers)) q |= it;
```
and instead of `auto` it is possible to write `Xapian::Query`.  Likewise 5 lines above.